src/main/java/com/example/tradingapp/trading/OkxOrderTracker.java

```diff
@@ -38,13 +38,28 @@ public void failed(String orderId) {
 
     @Override
     public void filled(String orderId, boolean fullFill) {
-
+        var newStatus = OrderStatus.FullyFilled;
+        changeStatus(newStatus, orderId, fullFill);
     }
 
     @Override
     public void canceled(String orderId) {
-        placedOrderIds.remove(orderId);
-        placedOrders.remove(orderId);
-        log.info("Removed from placed orders by ID: {}", orderId);
+        var newStatus = OrderStatus.Canceled;
+        changeStatus(newStatus, orderId, true);
+    }
+
+    private void changeStatus(OrderStatus newStatus, String orderId, boolean fullFill) {
+        if (placedOrders.get(orderId) == null) {
+            log.error("Can't change status to {} for Order({}), Order already removed", newStatus, orderId);
+        } else {
+            if (fullFill) {
+                OrderStatus oldStatus = placedOrders.get(orderId).getStatus();
+                placedOrders.get(orderId).setStatus(newStatus);
+
+                log.info("Order ({}) status change: {} -> {}", orderId, oldStatus, newStatus);
+            } else {
+                log.info("Order ({}) was partially filled", orderId);
+            }
```

Comment:
this actually can be moved out to partial fill handling logic.
then, we don't need to pass extra `fullFill` parameter